omod/src/main/java/org/openmrs/module/pihmalawi/rest/controller/DownloadDbRestController.java

```diff
@@ -28,14 +26,11 @@ public class DownloadDbRestController {
 
     private static String MANAGE_SYNC_PRIVILEGE = "Manage Synchronization";
 
-    @Autowired
-    private PublicKeyProvider publicKeyProvider;
     @GetMapping(value = "/rest/v1/pihmalawi/downloadDb")
     public ResponseEntity<byte[]> getDbBackup(@RequestHeader HttpHeaders headers) throws IOException {
-        List<String> signature = headers.get("X-Signature");
-        List<String> signatureInput = headers.get("Signature-Input");
+        List<String> macAddress = headers.get("MAC-Address");
         try {
-            if (signatureInput != null && signature != null && publicKeyProvider.verifySignature(signatureInput.get(0), signature.get(0))) {
+            if ( macAddress !=null && RestUtils.isMacAddressAllowed(macAddress.get(0))) {
```

Comment:
Presumably it would be safest to just have your RestUtils method take in a List<String> and call the method `anyMacAddressesAllowed(List<String>)`.  Then you don't have to worry about whether to just get the first one or not...

Reply:
I refactored this function. Thanks!